dask/array/blockwise.py

```diff
@@ -197,7 +199,21 @@ def blockwise(func, out_ind, *args, **kwargs):
                         "adjust_chunks values must be callable, int, or tuple")
     chunks = tuple(chunks)
 
-    return Array(graph, out, chunks, dtype=dtype)
+    # Attempt to construct meta, fallback to Array's dtype constructor if it's
+    # a basic type
+    try:
+        arrays = args[::2]
+        args_meta = [arg._meta if hasattr(arg, '_meta') else np.empty((0,)) for arg in arrays]
```

Comment:
I would expect this to be `else arg` instead of `else np.empty((0,))`.  What if the user provides an argument that is not an array, like a string?

```python
def f(a, b):
    if b == 'add':
        return a + 1
    elif b == 'sub':
        return a - 1
    else:
        raise ValueError(b)
    
blockwise(f, 'i', x, 'i', 'add', None)
```

Reply:
At first, I'm trying to get current tests to work, but yes, this is a valid concern. In fact, I've already seen some issues with that part of the code, I'll try to fix what I can before moving on to more complicated cases.

Reply:
In my [latest commit](https://github.com/dask/dask/pull/4543/commits/30c98c0d3c428945844340b91081d9cc81392406) I fixed the condition to `else arg`, as @mrocklin suggested.

I've added a second try/except to attempt passing `**kwargs` and if there's an exception, don't. I noticed the only case this happens is in the test that @mrocklin added here: https://github.com/dask/dask/pull/4543/files#diff-8f3045290928cb33f56d0b894e765b88R141

This happens because `blockwise()` is always called passing a `concatenate=True` in `map_blocks()`, as seen here https://github.com/dask/dask/blob/master/dask/array/core.py#L484, and `sparse.COO.from_numpy()` doesn't have this argument. What would be the correct solution here?